Website/admin/Security/SendPassword.ascx.cs

```diff
@@ -94,9 +95,8 @@ protected string RedirectURL
                         //return to the url passed to register
                         _RedirectURL = HttpUtility.UrlDecode(Request.QueryString["returnurl"]);
                         //redirect url should never contain a protocol ( if it does, it is likely a cross-site request forgery attempt )
-                        if (_RedirectURL.Contains("://") &&
-                            !_RedirectURL.StartsWith(Globals.AddHTTP(PortalSettings.PortalAlias.HTTPAlias),
-                                StringComparison.InvariantCultureIgnoreCase))
+                        if (_RedirectURL.Contains("://") &&
+							!Regex.Replace(_RedirectURL, @"^https*://", "", RegexOptions.IgnoreCase).StartsWith(PortalSettings.PortalAlias.HTTPAlias, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
the regex should be `^https?://` (so that only zero or one `s` is accepted), otherwise this change (in all three files) looks good.  I'd probably separate it into two `if` statements, so that it's clearer what's going on:

    if (_RedirectURL.Contains("://")) {
        var redirectWithoutProtocol = Regex.Replace(_RedirectURL, @"^https?://", string.Empty, RegexOptions.IgnoreCase);
        if (redirectWIthoutProtocol.StartsWith(PortalSettings.PortalAlias.HTTPAlias, StringComparison.InvariantIgnoreCase)) {
            _RedirectURL = "";
        }
    }